.github/workflows/helm-chart-ci.yaml

```diff
@@ -33,6 +33,28 @@ jobs:
       - name: Verify Docs updated
         run: ./helm-docs.sh
 
+  ensure-image-overrides:
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3.4.0
+        with:
+          fetch-depth: 0
+      - name: Check objects for images without overrides
+        run: |
+          set +e
+          set -o pipefail
+          # Look for image: defintions that are not templated. If we find none, exit is not 0 and we invert the error code to get the
+          # test to pass
+          grep -r image: charts/spire  | grep templates/ | grep -v 'image: {{ template "'
+          res=$?
+          if [ $res -eq 0 ]; then
+            echo The previously listed templates were found to be using statically defined images and not overridable ones. Please fix.
```

Comment:
Maybe write to the GITHUB_STEP_SUMMARY as markdown so it is nicely shown in the job summary.